cylc/flow/scripts/cat_log.py

```diff
@@ -568,13 +569,21 @@ def _main(
             workflow_id, point, task, submit_num
         )
 
+        job_log_present = (Path(local_log_dir) / "job.out").exists()
+        # For testing purposes
+        if not job_log_present:
+            LOG.debug("job.out not present, getting job log remotely")
+
         log_is_remote = (is_remote_platform(platform)
                          and (options.filename != JOB_LOG_ACTIVITY))
         log_is_retrieved = (platform['retrieve job logs']
                             and live_job_id is None)
         if (
+            # if the log file is not present locally
+            # (e.g. job is running, or job logs not retrieved)
+            not job_log_present
```

Comment:
The test failure is complaining that the error code from `cylc cat-log` has changed from `1` to `255`.

The `255` exit code is important because its the one SSH uses.

This test spotted a small logical flaw:
* If the `job.out` does not exist.
* And the job is being run locally.
* `cylc cat-log` will attempt to go remote.

I think this'll do it:

```suggestion
            (not job_log_present and log_is_remote)
```